test/t/thread/test_pool.cpp

```diff
@@ -48,9 +48,32 @@ TEST_CASE("number of threads in pool") {
 
 }
 
+TEST_CASE("number of threads in user-provided pool") {
+
+     SECTION("if zero requested, threads configured") {
```

Comment:
These should all be just TEST_CASEs without any SECTIONs. There is no common code here.